src/text-editor.coffee

```diff
@@ -2587,6 +2587,18 @@ class TextEditor extends Model
       maintainClipboard = true
     return
 
+
+  # Essential: For each selection, only copy hightlighted text. Line is coptied if no selected text and cursor is at the beginning of the line
+  copyOnlySelectedText: ->
+    maintainClipboard = false
+    for selection in @getSelectionsOrderedByBufferPosition()
+      if not selection.isEmpty()
+        selection.copy(maintainClipboard, true)
+      else if selection.isEmpty() and selection.cursor.isAtBeginningOfLine()
+        @copySelectedText()
```

Comment:
@izuzak I decided to use `copySelectedText()` because I thought reusing code would be nicer than inlining that part, but I hadn't considered that it would process all selections again - so I think you're right - it would be best to avoid doing the same thing twice. I'll make some changes :)